src/features/takeOtc/takeOtcActions.ts

```diff
@@ -74,3 +77,17 @@ export const takeOtcOrder = createAsyncThunk(
     console.log(swap);
   }
 );
+
+export const getTakenState = async (
+  order: FullOrder,
+  library: ethers.providers.Web3Provider
+) => {
+  const SwapContract = new Contract(
+    swapDeploys[order.chainId],
+    SwapInterface,
+    library.getSigner()
+  );
+
+  let tx = await SwapContract.nonceUsed(order.signerWallet, order.nonce);
+  return tx;
+};
```

Comment:
Actions are for `async` functions that mutate the store. This one doesn't mutate store so it should be a helper.